.github/workflows/check.yml

```diff
@@ -1,6 +1,8 @@
 name: check
 on:
   push:
+    branches: [master, 'test-me-*']
```

Comment:
Can we remove the test-me? and explain why this is needed?

Reply:
this is useful for testing CI without having to open a PR or push to master -- we have a similar setting in pytest

Reply:
But you shouldn't open PRs to core, you can enable the GHA for your fork if you want to test. And that's not sharing resources with upstream so I'm -1 on this.

Reply:
then I have to open junk PRs on my fork which I'd rather not do either -- this will let me just create branches to test them

Reply:
Why would you open a PR if you leave branches enabled for GHA, see https://github.com/gaborbernat/tox/actions

I still don't understand why this fixes it, and why we have double runs without this. I don't want you polluting the upstream repositories with stray branches. I understand for pre-commit.ci is a must, but for anything else, I'd rather not. Forks are the place to test what you want.

Reply:
release branches (as well as any changes made by github ui by any of the maintainers, such as typo fixes) will double-run everything because `push:` right now is blanket-enabled

Reply:
Why is that? What triggers the double-run?

Reply:
one run is triggered for `push:` and one run is triggered for `pull_request:`

Reply:
They're only the same build if you have rebased your Pr against the branch you're targeting, which is not always the case. In that case helpful to have two builds 🤔